update docker-inspect man page

hqhq opened the pull request:
- sort inspect out
- update output fields
- format output
- add doc about go template
- other minor fix

Signed-off-by: Qiang Huang h.huangqiang@huawei.com

moxiegirl commented on docs/man/docker-inspect.1.md:
```diff
@@ -30,69 +30,109 @@ each result.
 
 To get information on a container use it's ID or instance name:
```
`it's` > `its` can you change that while  you are in there.  Also, why was it necessary to change this example output so extensively?

hqhq replied:
Nothing special, I was reading this man page to figure out how I can write this `-f` and didn't get the answer. I think we should add something, then I found some format issues and the outdated inspect out, so just change them at one time.
The example output don't need to be updated in time, but it's too old now, I think it's time to refresh. :)

moxiegirl commented on docs/man/docker-inspect.1.md:
```diff
@@ -145,92 +175,92 @@ To get the IP address of a container use:
+        "VirtualSize": 613136466
     }]
 
 # HISTORY
```
Add a quick note to the history too please.

hqhq committed:
update docker-inspect man page

 - sort inspect out
 - update output fields
 - format output
 - add doc about go template
 - other minor fix

Signed-off-by: Qiang Huang <h.huangqiang@huawei.com>

hqhq commented:
@moxiegirl Updated, thanks.

moxiegirl commented:
Super! Thank you. LGTM

@thaJeztah @jamtur01 @fredlf @SvenDowideit

jamtur01 commented:
LGTM

thaJeztah commented:
LGTM
